CHANGELOG.md

```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Breaking changes
+
+* `jj [op] undo` and `jj op restore` used to take the operation to undo or
+  restore to as an argument to `-o/--operation`. It is now a positional
+  argument instead (i.e. `jj undo -o abc123` is now written `jj undo abc123`).
```

Comment:
Should it support both? How likely are people to get in the habit of writing `-o` for these commands, and it not being allowed being confusing?

Can undo take file arguments (or will it ever)? How would we specify those?

Reply:
> Should it support both? How likely are people to get in the habit of writing -o for these commands, and it not being allowed being confusing?

So far the commands accept only an operation and nothing else, and I'm not sure we'll ever change that. We can always add the option back later. By the way, I've considered doing that for e.g. `jj describe`, i.e. to make it accept a no-op `-r` flag.

> Can undo take file arguments (or will it ever)? How would we specify those?

I can't think what it would mean for it to take file arguments. For example, if you do a rebase (or pull from a remote), it doesn't seem to make much sense to undo that operation only for certain paths.